skills/declarative/calendarSkill/runtime/customaction/Actions/MSGraph/GetContacts.cs

```diff
@@ -102,7 +104,7 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
                     foreach (var email in person.ScoredEmailAddresses)
                     {
                         // If the email address isn't already included in the contacts list, add it
-                        if (!contactsResult.SelectMany(c => c.EmailAddresses).Contains(email.Address))
+                        if (!contactsResult.SelectMany(c => c.EmailAddresses).Contains(email.Address) && IsEmail(email.Address))
```

Comment:
This way of writing it is not optimal for large sets because we are asking the runtime to basically regenerate a stable list for every scored email address we have in the list which is an O(M*N) operation followed by another O(N) operation (the Contains()). 

I think we can make this code run faster, we can have a HashSet<T> of email addresses with the Contains() which turns it into an O(N) operation, then an O(1).

Reply:
good point, updated the code.